### PR TITLE
Update go to 1.18, k8s to 0.24.2 

### PR DESCRIPTION
**Pull request by ranakan19**

### CODE REVIEW

**Review by xcoulon — APPROVED** [review]

**Review by xcoulon — APPROVED** [review]

File: config/crd/bases/toolchain.dev.openshift.com_idlers.yaml [hunk 1]
```diff
@@ -118,9 +118,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
```
Comment (xcoulon) [inline config/crd/bases/toolchain.dev.openshift.com_idlers.yaml:121]: I wonder why all the `status` have been removed by `controller-gen` 🤔

Comment (ranakan19) [reply config/crd/bases/toolchain.dev.openshift.com_idlers.yaml:121]: @xcoulon yes, i was thinking the same. For reference - https://github.com/kubernetes-sigs/controller-tools/pull/630 - explains a bit more about the choice of removing status.

File: go.mod [hunk 2]
```diff
@@ -91,13 +91,18 @@ require (
...
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-go 1.17
+go 1.18
+
+replace (
+	github.com/codeready-toolchain/api v0.0.0-20221121081259-4f672e808b89 => github.com/ranakan19/api v0.0.0-20221201091806-57fc665ce345
+	github.com/codeready-toolchain/toolchain-common v0.0.0-20221117081323-252a38460fa1 => github.com/ranakan19/toolchain-common v0.0.0-20221201182049-e9ceb160687f
```
Comment (mfrancisc) [inline go.mod:107]: friendly reminder!

**Review by mfrancisc — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**